core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -114,70 +145,62 @@ class CoarseCookSchedulerBackend(
     .build()
 
   private[this] val jobListener = new CJobListener {
+
     // These are called serially so don't need to worry about race conditions
     def onStatusUpdate(job: Job) {
       val isCompleted = job.getStatus == Job.Status.COMPLETED
-      val isAborted = abortedJobIds.contains(job.getUUID)
+      val isAborted = AbortedJobUUIDs.contains(job.getUUID)
 
       if (isCompleted) {
-        executorsRequested -= 1
-        abortedJobIds -= job.getUUID
-        jobIds -= job.getUUID
+        nonCompletedJobUUIDs.remove(job.getUUID)
+        AbortedJobUUIDs.remove(job.getUUID)
 
         if (isAborted) {
-          logInfo(s"Job ${job.getUUID} has been successfully aborted.")
+          logInfo(
+            s"Job ${job.getUUID} has been successfully aborted per request.")
         }
 
         if (!job.isSuccess && !isAborted) {
-          totalFailures += 1
-          logWarning(s"Job ${job.getUUID} has died. " +
-            s"Failure ($totalFailures/${schedulerConf.getMaximumExecutorFailures})")
-          jobIds -= job.getUUID
-          if (totalFailures >= schedulerConf.getMaximumExecutorFailures) {
-            // TODO should we abort the outstanding tasks now?
-            logError(
-              s"We have exceeded our maximum failures " +
-                s"($schedulerConf.getMaximumExecutorFailures)" +
-                "and will not relaunch any more tasks")
-          }
+          logWarning(s"Job ${job.getUUID} has died unintentionally.")
```

Comment:
I think we already get this logging from executorDisconnect below. We can probably remove this. Same thing with the aborted part above.